modules/unsupported/solr/src/test/java/org/geotools/data/solr/TestsSolrUtils.java

```diff
@@ -0,0 +1,370 @@
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.solr;
+
+import org.apache.solr.client.solrj.SolrRequest;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.request.UpdateRequest;
+import org.apache.solr.client.solrj.request.schema.FieldTypeDefinition;
+import org.apache.solr.client.solrj.request.schema.SchemaRequest;
+import org.apache.solr.client.solrj.response.SolrResponseBase;
+import org.apache.solr.common.SolrInputDocument;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.InputStream;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This class contains helper methods typically used when dealing with
+ * Apache Solr integration tests.
+ */
+public final class TestsSolrUtils {
+
+    /**
+     * Instantiates a new Apache Solr client using the provided base URL. The provided
+     * base URL should point to a specific Solr core.
+     *
+     * @param baseUrl base URL of the Apache Solr instance, http://localhost:8983/solr/core
+     * @return Apache Solr client instance
+     */
+    public static HttpSolrClient instantiateClient(String baseUrl) {
+        HttpSolrClient client = new HttpSolrClient.Builder()
+                .withBaseSolrUrl(baseUrl)
+                .allowCompression(true)
+                .build();
+        // we can use low timeouts values for tests
+        client.setConnectionTimeout(5000);
+        client.setSoTimeout(5000);
+        client.setFollowRedirects(true);
+        return client;
+    }
+
+    /**
+     * Performs an update request extracting the necessary Solr documents from the
+     * provided stream. It is expected that the stream provides Solr documents
+     * encoded in the XML format:
+     * <p>
+     * <pre>
+     * {@code
+     * (...)
+     * <doc>
+     *  <field name="id">1</field>
+     *  <field name="status_s">active</field>
+     *  <field name="vendor_s">D-Link</field>
+     *  (...)
+     * </doc>
+     * }
+     * </pre>
+     *
+     * @param client Sorl client to use, should already point to the desired core
+     * @param xml    stream contain the Solr documents encoded in XML
+     */
+    public static void updateRequest(HttpSolrClient client, InputStream xml) {
```

Comment:
Maybe "runUpdate". At first I thought the code was going to modify a request but saw nothing returned from the method...

Reply:
I agree indeed the method name is misleading, I will also update all the other methods names that are actually running a request to contain the term run in its name.